File: 백준/Silver/1002. 터렛/터렛.c
#include <stdio.h>
#include <math.h>

int turret();

int main()
{
	int input;

	scanf("%d", &input);
	for (int i = 0; i < input; i++)
	{
		printf("%d\n", turret());
	}

	return 0;
}

int turret()
{
	int info[6];
	int len[3];
	int longest, other[2];

	for (int i = 0; i < 6; i++)
	{
		scanf("%d", &info[i]);
		getchar();
	}
	len[0] = pow(info[0] - info[3], 2) + pow(info[1] - info[4], 2);
	len[1] = pow(info[2], 2);
	len[2] = pow(info[5], 2);

	if (len[0] >= len[1] && len[0] >= len[2])
	{
		longest = len[0];
		other[0] = len[1];
		other[1] = len[2];
	}
	else if (len[1] >= len[0] && len[1] >= len[2])
	{
		longest = len[1];
		other[0] = len[0];
		other[1] = len[2];
	}
	else if (len[2] >= len[0] && len[2] >= len[1])
	{
		longest = len[2];
		other[0] = len[0];
		other[1] = len[1];
	}

	if (sqrt(longest) < sqrt(other[0]) + sqrt(other[1]))
		return 2;
	else if (info[0] == info[3] && info[1] == info[4] && info[2] == info[5])
		return -1;
	else if (sqrt(longest) == sqrt(other[0]) + sqrt(other[1]))
		return 1;
	else
		return 0;
}

File: 백준/Silver/1002. 터렛/터렛.cs
using System;

namespace AlgorithmStudy
{
    class boj1002
    {
        public static void Main(string[] args)
        {
            int n = int.Parse(Console.ReadLine());

            for (int i = 0; i < n; i++)
            {
                string[] input = Console.ReadLine().Split();
                int x1 = int.Parse(input[0]), y1 = int.Parse(input[1]), r1 = int.Parse(input[2]), x2 = int.Parse(input[3]), y2 = int.Parse(input[4]), r2 = int.Parse(input[5]);
                int len = (int)(Math.Pow(x2 - x1, 2) + Math.Pow(y2 - y1, 2));
                if (x1 == x2 && y1 == y2)
                {
                    if (r1 == r2)
                        Console.WriteLine("-1");
                    else
                        Console.WriteLine("0");
                }
                else
                {
                    if (len == Math.Pow(r1 + r2, 2) || (r1 > r2 ? Math.Pow(r1 - r2, 2) : Math.Pow(r2 - r1, 2)) == len)
                        Console.WriteLine("1");
                    else if (len > Math.Pow(r1 + r2, 2) || (r1 > r2 ? Math.Pow(r1 - r2, 2) : Math.Pow(r2 - r1, 2)) > len)
                        Console.WriteLine("0");
                    else if (len < Math.Pow(r1 + r2, 2))
                        Console.WriteLine("2");
                }
            }
        }
    }
}
